src/components/Prescription/Prescription.js

```diff
@@ -9,23 +9,29 @@ const Prescription = () => {
   const [show, setShow] = useState({});
   const [prescription, setPrescription] = useState('');
   const { id }  = useParams();
+  const navigate = useNavigate()
 
   useEffect(() => {
-    getPrescription(id)
-      .then(data => {
-        setPrescription(data.prescription);
-        getShowById(data.prescription.showID)
-          .then(data => {
-            const { name, officialSite, image } = data;
-            setShow({
-              name: name,
-              officialSite: officialSite,
-              image: image.medium
-            })
-          })    
-          .catch(error => console.log(error));      
-        })
-        .catch(error => console.log(error));
+    if(parseInt(id) === NaN) {
+      navigate('/prescription/prescription-not-found')
+      return 
+    } else {
+      getPrescription(id)
+        .then(data => {
+          setPrescription(data.prescription);
+          getShowById(data.prescription.showID)
+            .then(data => {
+              const { name, officialSite, image } = data;
+              setShow({
+                name: name,
+                officialSite: officialSite,
+                image: image.medium
+              })
+            })    
+            .catch(error => navigate('/prescription/prescription-not-found'));      
+          })
+          .catch(error => navigate('/prescription/prescription-not-found'));
```

Comment:
I see that you are routing the user when a prescription not found error is returned. I think this is smart to use the error, but there still needs to be a 404 page when a  user types in an endpoint that does not exist. I think we also need to add something for if a user tries to go to `/garbage-town-usa/45` for example.